packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -409,6 +411,54 @@ class AndroidSdk {
     return null;
   }
 
+  // Returns the version of java in the format \d(.\d)+(.\d)+
+  // Returns null if version not found.
+  String? getJavaVersion({
+    required AndroidStudio? androidStudio,
+    required FileSystem fileSystem,
+    required OperatingSystemUtils operatingSystemUtils,
+    required Platform platform,
+    required ProcessUtils processUtils,
+  }) {
+    final String? javaBinary = findJavaBinary(
+      androidStudio: androidStudio,
+      fileSystem: fileSystem,
+      operatingSystemUtils: operatingSystemUtils,
+      platform: platform,
+    );
+    if (javaBinary == null) {
+      return null;
+    }
+    final RunResult result = processUtils.runSync(
+      <String>[javaBinary, '--version'],
+      environment: sdkManagerEnv,
+    );
+    if (result.exitCode != 0) {
+      globals.printTrace(
+          'java --version failed: exitCode: ${result.exitCode} stdout: ${result.stdout} stderr: ${result.stderr}');
+      return null;
+    }
+    return parseJavaVersion(result.stdout);
+  }
+
+  // Java versions have 2 formats
+  @visibleForTesting
+  static String? parseJavaVersion(String? rawVersionOutput) {
```

Comment:
Optional nit: You could make the parameter non-nullable, and the call it with `restult.stdout ?? ''`. That avoids the need to have special null handling in the parsing method just because of how it happens to be called.